SU2_CFD/src/driver_structure.cpp

```diff
@@ -6952,3 +7113,313 @@ void CDiscAdjFSIStatDriver::Postprocess(unsigned short ZONE_FLOW,
 
 
 }
+
+CMultiPhysicalZonesDriver::CMultiPhysicalZonesDriver(char* confFile,
+                       unsigned short val_nZone,
+                       unsigned short val_nDim,
+                       SU2_Comm MPICommunicator) : CDriver(confFile,
+                                                          val_nZone,
+                                                          val_nDim,
+                                                          MPICommunicator) { }
+
+CMultiPhysicalZonesDriver::~CMultiPhysicalZonesDriver(void) { }
+
+void CMultiPhysicalZonesDriver::Run() {
```

Comment:
I really like the idea of abstracting the multiphysics driver to the point that it can (in theory) include almost any set of physics in any zone with modular coupling at the interfaces. This is the right vision, I think, and we've been working in this direction. As you already know :) it will also make the construction of multiphysics adjoints with AD even more straightforward. Looking forward to the discrete adjoint of this when you have time to push it (and future multiphysics adjoints). Nice job here.

Reply:
It's ready. I just wanted to avoid writing the clean version of "CDiscAdjMultiphysicsZonalDriver" before this pull request as you may suggest/request changes (like the naming of the thing :D).

My local discrete adjoint driver (so far only for CHT, but following this approach arbitrary couplings should be no problem, as long as we can formulate them as a sequential one to fit the taping/pseudo-time framework) gives 100% accurate results.